doc/sphinx/source/user_guide2/preprocessor.inc

```diff
@@ -28,6 +29,76 @@ The _time.py module contains the following preprocessor functions:
 * extract_month: Extract only the times that occur within a specific month.
 * time_average: Take the weighted average over the time dimension.
 * seasonal_mean: Produces a mean for each season (DJF, MAM, JJA, SON)
+* annual_mean: Produces an annual or decadal mean.
+
+1. extract_time
+---------------
+
+This function subsets a dataset between two points in times. It removes all
+times in the dataset before the first time and after the last time point.
+The required arguments are relatively self explanatory:
+
+* start_year
+* start_month
+* start_day
+* end_year
+* end_month
+* end_day
+
```

Comment:
I think it would be good to add links to the function documentation, i.e. add something like
```
See also :func:`esmvaltool.preprocessor.extract_time`.
```
below every function.

I think that we should be careful that we not duplicate too much what is already documented in the code. It would be better to describe here how the preprocessor works when used from the recipe. E.g. provide information on what values are automatically filled from the recipe, e.g. for this function, start_year and end_year (= dataset end_year + 1) are filled in from the dataset section, if no specific values are provided in the recipe.

Reply:
I've added the links to the API. 

I'd like to think that the code in the API includes a description which would be useful for people who are happy to read and write python code, but that the documentation in the Sphinx build is more geared towards people writing recipes.

Reply:
Yes, I think so too. Could you add this information, to make it useful for recipe users?
> It would be better to describe here how the preprocessor works when used from the recipe. E.g. provide information on what values are automatically filled from the recipe, e.g. for this function, start_year and end_year (= dataset end_year + 1) are filled in from the dataset section, if no specific values are provided in the recipe.